pkgs/development/python-modules/extractcode/libarchive.nix

```
@@ -0,0 +1,42 @@
+{ lib
+, fetchFromGitHub
+, buildPythonPackage
+, pkgs
+}:
+buildPythonPackage rec {
+  pname = "extractcode-libarchive";
+  version = "21.4.4";
+
+  src = fetchFromGitHub {
+    owner = "nexB";
+    repo = "scancode-plugins";
+    rev = "v${version}";
+    sha256 = "xnUGDMS34iMVMGo/nZwRarGzzbj3X4Rt+YHvvKpmy6A=";
+  };
+
+  postPatch = ''
+    cd builtins/extractcode_libarchive-linux
+  '';
+
+  preBuild = ''
+    pushd src/extractcode_libarchive/lib
+
+    cp ${lib.getLib pkgs.libarchive}/lib/libarchive.so libarchive.so
+    cp ${lib.getLib pkgs.libb2}/lib/libb2.so libb2-la3511.so.1
+    cp ${lib.getLib pkgs.bzip2}/lib/libbz2.so libbz2-la3511.so.1.0
+    cp ${lib.getLib pkgs.expat}/lib/libexpat.so libexpat-la3511.so.1
+    cp ${lib.getLib pkgs.lz4}/lib/liblz4.so liblz4-la3511.so.1
+    cp ${lib.getLib pkgs.lzma}/lib/liblzma.so liblzma-la3511.so.5
+    cp ${lib.getLib pkgs.zlib}/lib/libz.so libz-la3511.so.1
+    cp ${lib.getLib pkgs.zstd}/lib/libzstd.so libzstd-la3511.so.1
```

```suggestion
    cp ${lib.getLib libarchive}/lib/libarchive.so libarchive.so
    cp ${lib.getLib libb2}/lib/libb2.so libb2-la3511.so.1
    cp ${lib.getLib bzip2}/lib/libbz2.so libbz2-la3511.so.1.0
    cp ${lib.getLib expat}/lib/libexpat.so libexpat-la3511.so.1
    cp ${lib.getLib lz4}/lib/liblz4.so liblz4-la3511.so.1
    cp ${lib.getLib lzma}/lib/liblzma.so liblzma-la3511.so.5
    cp ${lib.getLib zlib}/lib/libz.so libz-la3511.so.1
    cp ${lib.getLib zstd}/lib/libzstd.so libzstd-la3511.so.1
```

Please inherit when pythonPackages with equal names exist.